test/test-addon.js

```diff
@@ -25,4 +22,44 @@ test('build simple addon', function (t) {
   })
   proc.stdout.setEncoding('utf-8')
   proc.stderr.setEncoding('utf-8')
+}
+
+test('build simple addon', function (t) {
+  t.plan(3)
+  // Set the loglevel otherwise the output disappears when run via 'npm test'
+  runCmd(t, [nodeGyp, 'rebuild', '-C', addonPath, '--loglevel=verbose'])
+})
+
+test('build simple addon in path with non-ascii characters', function (t) {
+  t.plan(6)
+
+  var data, config, nodeDir, testNodeDir
+  var configPath = path.join(addonPath, 'build', 'config.gypi')
+
+  try {
+    data = fs.readFileSync(configPath, 'utf8')
+  } catch (err) {
+    t.error(err)
+    return
+  }
+  config = JSON.parse(data.replace(/\#.+\n/, ''))
+  nodeDir = config.variables.nodedir
+  // Create path with non-ascii characters
+  testNodeDir = path.join(addonPath, '非英文字符')
+  try {
+    fs.symlinkSync(nodeDir, testNodeDir, 'dir')
+  } catch (err) {
+    switch (err.code) {
+      case 'EEXIST': break
+      case 'EPERM':
+        t.error(err, 'Please try to running console as an administrator')
+        return
+      default:
+        t.error(err)
+        return
+    }
+  }
+  runCmd(t, [nodeGyp, 'configure', '-C', addonPath, '--loglevel=verbose',
+              '-nodedir=' + testNodeDir])
+  runCmd(t, [nodeGyp, 'build', '-C', addonPath, '--loglevel=verbose'])
```

Comment:
@refack I have tried to use the `rebuild` command before, but there will be errors:

```
...
gyp verb clean removing "build" directory
gyp ERR! clean error
gyp ERR! stack Error: EPERM: operation not permitted, unlink 'F:\repos\GitHub\node-gyp\test\node_modules\hello_world\build\Release\hello.node'
...
```

how to solve this problem? a temporary solution is to use `configure` and `build` commands.